src/dotnet/UI.Blazor/Components/Navbar/NavbarContent.razor

```diff
@@ -33,3 +35,20 @@
     </NavbarGroup>
 }
 
+@code {
+    [Inject] private NavbarUI NavbarUI { get; init; } = null!;
+
+    protected override void OnInitialized() {
+        base.OnInitialized();
+        NavbarUI.ActiveGroupChanged += NavbarUIOnActiveGroupChanged;
+    }
+
+    public override ValueTask DisposeAsync() {
+        NavbarUI.ActiveGroupChanged -= NavbarUIOnActiveGroupChanged;
+        return base.DisposeAsync();
+    }
+
+    private void NavbarUIOnActiveGroupChanged(object? sender, EventArgs e) {
+        StateHasChanged();
```

Comment:
Is this necessary? If I am not mistaken, by default components call `StateHasChanged` after any event. I intentionally suppressed this in `StatefulComponentBase` though, coz typically this isn't desirable there, but you can enable default behavior by setting `TriggerStateHasChangedOnEvent` to true.

Now, if you want to disable this on per event basis, here is how you can do this now: https://github.com/Actual-Chat/actual-chat/commit/6bfad03f3bc89e728566141f36ec41816336fd3d#diff-5aab49c9c1934d191ea4b3e892751bf063e75c4eaae33f1e848509cc7d5ec5e3R8